include/wayland/mir/wayland/wayland_base.h

```diff
@@ -215,6 +215,10 @@ class Global
     wl_global* const global;
 };
 
+/// Calls wl_client_post_implementation_error() if available
+[[gnu::format(printf, 2, 3)]]
+void post_implementation_error(wl_client* client, char const* fmt, ...);
+
 void internal_error_processing_request(wl_client* client, char const* method_name);
```

Comment:
The naming and functionality of these functions seems a bit unclear:

1. the "internal" function calls the not-"internal" function, it is usually the other way around.
2. everywhere we call `post_implementation_error()`, we also log the error (which is the difference in functionality)

It looks to me that the _only_ reason for `post_implementation_error()` to exist is the absence of an active exception when called from `WindowWlSurfaceRole::surface_destroyed()`.